testnet/stacks-node/src/neon_node.rs

```diff
@@ -811,6 +811,19 @@ fn spawn_miner_relayer(
                         event_dispatcher.process_new_mempool_txs(net_receipts.mempool_txs_added);
                     }
 
+                    let num_unconfirmed_microblock_tx_receipts = net_receipts.processed_unconfirmed_state.receipts.len();
+                    if num_unconfirmed_microblock_tx_receipts > 0 {
+                        let (canonical_consensus_hash, canonical_block_hash) =
+                            SortitionDB::get_canonical_stacks_chain_tip_hash(
```

Comment:
I think you can just use this here:

```rust
let canonical_tip = chainstate
   .as_ref()
   .map(|unconfirmed_state| unconfirmed_state.confirmed_chain_tip.clone())
   .unwrap_or(StacksBlockId([0u8; 32]));

if canonical_tip != StacksBlockId([0u8; 32]) {
   event_dispatcher.process_new_microblocks(canonical_tip, net_receipts.processed_unconfirmed_state);
}
else {
   // this really should be unreachable
   warn!("OOPS: unconfirmed state is uninitialized but there are microblock events");
}
```

The `unwrap_or()` case should never be reached if `new_unconfirmed_microblock_tx_receipts.len() > 0`, since the unconfirmed state will always be initialized and the `confirmed_chain_tip` will correspond to the Stacks chain tip on top of which the unconfirmed microblocks are built.

Reply:
My question about this change is that it seems like `process_new_microblocks` will be called regardless of the value of `net_receipts.processed_unconfirmed_state.receipts.len()`. I can make this change, but I guess I should add a check in `process_new_microblocks` to ensure the length of the receipts is non-zero before executing any logic?

Reply:
Oh, I was only talking about removing the call to `SortitionDB::get_canonical_stacks_chain_tip_hash()`.  The above code would run only if `net_receipts.processed_unconfirmed_state.receipts.len() > 0`.